sktime/forecasting/callbacks/mlflow.py

```diff
@@ -0,0 +1,252 @@
+    ...         y=y,
+    ...         cv=cv,
+    ...         strategy="update",
+    ...         callbacks=[mlflow_callback]
+    ...     )
+
+    # Set MLflow callback for nested runs:
+    >>> mlflow_callback = MLFlowCallback(experiment_id=experiment_id, nested=True)
+
+    # Start parent run if nested is set to True:
+    >>> with mlflow.start_run(run_name="parent_run"):
+    ...     # Evaluate NaiveForecaster with MLFlowCallback
+    ...     results_naive = evaluate(
+    ...         forecaster=NaiveForecaster(strategy="mean", sp=3),
+    ...         y=y,
+    ...         cv=cv,
+    ...         strategy="update",
+    ...         callbacks=[mlflow_callback],
+    ...         return_data=True
+    ...     )
+    ...
+    ...     # Evaluate PolynomialTrendForecaster with MLFlowCallback
+    ...     results_polynomial = evaluate(
+    ...         forecaster=PolynomialTrendForecaster(degree=2),
+    ...         y=y,
+    ...         cv=cv,
+    ...         strategy="update",
+    ...         callbacks=[mlflow_callback],
+    ...         return_data=True
+    ...     )
+    """
+
+    def __init__(
+        self,
+        forecaster: Optional[BaseForecaster] = None,
+        scores: Optional[list] = None,
+        tracking_uri: str = None,
+        run_name: Optional[str] = None,
+        experiment_id: str = None,
+        nested: bool = False,
+    ):
+        super().__init__()
+        import mlflow
+
+        self.experiment_id = experiment_id
+        self.nested = nested
+        self.tracking_uri = tracking_uri
+        self._forecaster = forecaster
+        self.score_metrics = scores
+        self.runs = []
+        self.run_name = run_name
+
+        if tracking_uri:
+            mlflow.set_tracking_uri(tracking_uri)
+
+    @property
+    def forecaster(self):
+        """Forecaster being evaluated."""
+        return self._forecaster
+
+    @forecaster.setter
+    def forecaster(self, forecaster):
+        self._forecaster = forecaster
+        if not self.run_name:
+            self.run_name = forecaster.__class__.__name__
+
+    def _reset(self):
+        """Reset relevant attributes (forecaster, run_name, score_metrics).
+
+        It is expected that the same MLFLowCallback object can be reused multiple times.
+        Therefore certain variables should be reset everytime that the `evaluate`
+        function is finished.
+        """
+        self.run_name = None
+        self.forecaster = None
+        self.score_metrics = []
+
+    def on_iteration(self, iteration, x, result, update=None):
+        """Start MLFlow run or open existing run."""
+        import mlflow
+
+        _, (y_train, y_test, X_train, X_test) = x
+        scores = {}
+        y_pred = result["y_pred"]
+        for score in self.score_metrics:
+            scores[f"{score.name}"] = result[f"test_{score.name}"].iloc[0]
+            mlflow.log_metric(
+                f"{score.name}", value=result[f"test_{score.name}"], step=iteration
+            )
```

Comment:
Suggestion: can we use [`log_metrics`](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.log_metrics) to log all scores simultaneously instead of one by one?